src/foam/dao/EasyDAO.js

```diff
@@ -236,6 +226,11 @@ if ( getNSpec() != null &&
   logger.warning("EasyDAO", getNSpec().getName(), "Served DAO should be Authenticated, Authorized, or ReadOnly");
 }
 
+if ( getPermissioned() &&
```

Comment:
should authorization go before permissioned property to avoid redundant checks in case authorizationDAO doesn't allow to read at ll?

Reply:
That can cause problems where `PermissionedPropertyDAO` might reset a property to its default value before the authorization logic called by `AuthorizationDAO` can access it.

Say a model `Foo` has a permissioned property `bar` of type `Boolean`, where the default value is false. Lets consider a `find`, for example. We have some instance of `Foo` we're trying to find and we know the real value of `bar` is `true`. `AuthorizationDAO` happens before `PermissionedPropertyDAO` and calls `super.find_` to get the object before running the authorization checks on it. That call to `super.find_` will trigger `PermissionedPropertyDAO.find_`, which clears `bar` back to its default value of false. The `authorizeOnRead` code then gets executed, but it contains `if ( getBar() ) { ... }`. The problem is that `getBar()` will always be false since it has already been reset.

---

There are other ways to solve this problem, but moving the `PermissionedPropertyDAO` outside of `AuthorizationDAO` is one of them.